app/controllers/CyaCreateRecordController.scala

```diff
@@ -41,29 +43,43 @@ class CyaCreateRecordController @Inject() (
   val controllerComponents: MessagesControllerComponents,
   view: CyaCreateRecordView,
   goodsRecordConnector: GoodsRecordConnector,
+  ottConnector: OttConnector,
+  sessionRepository: SessionRepository,
   auditService: AuditService
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport
     with Logging {
 
-  def onPageLoad(): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
+  def onPageLoad(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
     GoodsRecord.build(request.userAnswers, request.eori) match {
       case Right(_)     =>
-        val list = SummaryListViewModel(
-          rows = Seq(
-            TraderReferenceSummary.row(request.userAnswers),
-            UseTraderReferenceSummary.row(request.userAnswers),
-            GoodsDescriptionSummary.row(request.userAnswers),
-            CountryOfOriginSummary.row(request.userAnswers),
-            CommodityCodeSummary.row(request.userAnswers)
-          ).flatten
-        )
-        Ok(view(list))
-      case Left(errors) => logErrorsAndContinue(errors)
+        request.userAnswers.get(CountriesQuery) match {
+          case Some(countries) => Future.successful(displayView(request.userAnswers, countries))
```

Comment:
how come we need to retrieve the country list for CYA? Can't we just retrieve the CountryOfOriginPage value or am I oversimplifying it?

Reply:
No because we need to check that it is from the required list of countries for error checking